app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/MapScreen.kt

```diff
@@ -81,6 +81,7 @@ fun MapScreen(navigationActions: NavigationActions, viewModel: MapViewModel) {
     LaunchedEffect(true) {
       val intent = Intent(context, LocationUpdatesService::class.java)
       context.startService(intent)
+      viewModel.getProfileOfCurrentUser()
     }
 
     DisposableEffect(Unit) {
```

Comment:
When does this get executed? Usually, when we have the map open, and walk around, we want to add and retrieve songs from beacons continuously, which I don't really see happening here. Can you clarify?

Reply:
This is executed at the launch of the application, once the user login we have the map, and then we retrieve the profile. To see where we add and drop the songs contiuously you can look below in the disposableEffect, we call createLocationReceiver, and here, we call the corresponding functions. This function was at the beginning looking for the POI near a user every 30seconds, but it can also do other things, as retrieving and droping songs. Also I thought that we only need to retrieve the user profile once(at the beginning), but maybe there is an other reason why we would want to fetch it more often?